sklearn/metrics/cluster/expected_mutual_info_fast.pyx

```diff
@@ -28,8 +28,8 @@ def expected_mutual_information(contingency, int n_samples):
     #cdef np.ndarray[int, ndim=2] start, end
     R, C = contingency.shape
     N = <DOUBLE>n_samples
-    a = np.sum(contingency, axis=1).astype(np.int32)
-    b = np.sum(contingency, axis=0).astype(np.int32)
+    a = np.ravel(contingency.sum(axis=1).astype(np.int32))
```

Comment:
Because it can now be a sparse matrix and the sum over the rows of a sparse matrix is a sparse matrix with a single column instead of a 1D array.